gorm/utilities.go

```diff
@@ -40,6 +41,62 @@ func HandleFieldPath(ctx context.Context, fieldPath []string, obj interface{}) (
 	return dbPath, "", nil
 }
 
+//HandleJSONFiledPath translate field path to JSONB path for postgres jsonb
+func HandleJSONFieldPath(ctx context.Context, fieldPath []string, obj interface{}, values ...string) (string, string, error) {
+	operator := "#>>"
+	if isRawJSON(values...) {
+		operator = "#>"
+	}
+
+	dbPath, err := fieldPathToDBName(fieldPath[:1], obj)
+	if err != nil {
+		switch err.(type) {
+		case *EmptyFieldPathError:
+			return "", "", err
+		default:
+			dbPath = fieldPath[0]
+		}
+	}
+
+	if len(fieldPath) == 1 {
+		return dbPath, "", nil
+	}
+
+	return fmt.Sprintf("%s %s '{%s}'", dbPath, operator, strings.Join(fieldPath[1:], ",")), "", nil
+}
+
+func isRawJSON(values ...string) bool {
+	if len(values) == 0 {
+		return false
+	}
+
+	for _, v := range values {
+		if !strings.HasPrefix(v, "{") || !strings.HasSuffix(v, "}") {
```

Comment:
consider move to regexp checking. Also think if cases with leading/trailing spaces are valid